### PR TITLE
SRCH-4961 Data Migration for Launch Support - ALL sites - Image Content settings

### PR DESCRIPTION
**Pull request by krbhavith**
## Summary
- Update new SERP image content settings data with old content settings data
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: app/models/image_content_updater.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      migrate_paperclip_image_to_active_storage(affiliate)
+      update_alt_text(affiliate)
+      affiliate.save
+    end
+  
+    def migrate_paperclip_image_to_active_storage(affiliate)
+      return if affiliate.mobile_logo_file_name.blank?
+  
+      logo_url = affiliate.send(:mobile_logo).url.sub(Rails.env, 'production').gsub(' ', '%20')
+      affiliate.header_logo.attach(io: Tempfile.open(logo_url),
```
Comment (krbhavith) [inline app/models/image_content_updater.rb:36]: @lorawoodford here I am following the same process what is mentioned in [documentation](https://github.com/thoughtbot/paperclip/blob/main/MIGRATING.md#moving-files-on-a-remote-host-s3-azure-storage-gcs-etc) to migrate the images.

File: app/models/image_content_updater.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    def update_affiliate_image_content(affiliate)
+      migrate_paperclip_image_to_active_storage(affiliate)
+      update_alt_text(affiliate)
+      affiliate.save
+    end
+  
+    def migrate_paperclip_image_to_active_storage(affiliate)
+      return if affiliate.mobile_logo_file_name.blank?
+  
+      logo_url = affiliate.send(:mobile_logo).url.sub(Rails.env, 'production').gsub(' ', '%20')
```
Comment (lorawoodford) [inline app/models/image_content_updater.rb:35]: @krbhavith This seems unnecessarily complex.  What stops us from doing `affiliate.mobile_logo.url`?

Comment (krbhavith) [reply app/models/image_content_updater.rb:35]: Right we can use `affiliate.mobile_logo_url`

File: app/models/image_content_updater.rb (hunk 1 above)
Comment (krbhavith) [reply app/models/image_content_updater.rb:36]: Still `active_record_attachment` and `active_record_blob` is created properly with all the values except the `byte_size`, size is always 0.

**Review by lorawoodford — COMMENTED** [review]
@krbhavith This is definitely on the right track.  I may be paranoid, but I also think it would be wise to add some additional feedback/confirmation steps and logging, particularly I'm wondering about:

- Should we printing to the console what Paperclip bucket we're working against before continuing the task?  We want to be very careful when we're testing that we're working against the `dgsearch-images-dev` bucket, but want to be equally assured that we're working against `dgsearch-images` when we run it for real in production.
- Can we anticipate any additional logging needs beyond just success/fail?  Do we want to know how long this takes?
- How many production affiliates have a logo?  If there's a sizable number lacking a logo, we'd likely be better off filtering them out in the initial query before getting into the `migrate_paperclip_image_to_active_storage` method.

Again, your call to make here.  Once the codeclimate issues and other comment is addressed, I'm happy to do a formal review.  Ping me with questions.

File: app/models/image_content_updater.rb (hunk 1 above)
Comment (lorawoodford) [reply app/models/image_content_updater.rb:36]: @krbhavith Looks like all this need is changing

```
affiliate.header_logo.attach(io: Tempfile.open(logo_url),
```

to:
```
affiliate.header_logo.attach(io: URI.open(logo_url),
```

**Review by lorawoodford — APPROVED** [review]
@krbhavith This LGTM.  As a nitpick, it should be fairly straightforward to stub a response from S3 so we don't get those meaningless/error-filled VCR cassettes, but I don't see that as a reason to delay this any further.  If you have time, though, it'd be a nice to do.

**Pull request merged**